pai-management/paiLibrary/common/file_handler.py

```diff
@@ -69,3 +69,39 @@ def file_delete(file_path):
 def create_folder_if_not_exist(folder_path):
     if not os.path.exists(folder_path):
         os.makedirs(folder_path)
+
+
+def add_deploy_rule_to_yaml(service_conf, src_yaml, dst_yaml):
+    
+    config = dict()
+    service_deploy_kind_list = ['DaemonSet', 'Deployment', 'StatefulSets', 'Pod']
+
+    with open(src_yaml, 'r') as f:
+        config = yaml.load(f)
+
+        # judge whether it's a service deploy file, eg. exclude configmap
+        if 'kind' in config and config['kind'] in service_deploy_kind_list:
+            match_expressions_arr = []
+
+            deploy_rules =service_conf['deploy-rules']
+            for operator, label in deploy_rules.items():
+                match_expression = dict()
+                if operator.lower() == 'in':   
+                    match_expression['operator'] = 'In'
+                if operator.lower() == 'notin':
+                    match_expression['operator'] = 'NotIn'
+                              
+                match_expression['key'] = label
+                match_expression['values'] = ['true']
+                match_expressions_arr.append(match_expression)
+
+            config['spec']['template']['spec']['affinity'] = {'nodeAffinity': \
+                {'requiredDuringSchedulingIgnoredDuringExecution': {'nodeSelectorTerms': \
+                [{'matchExpressions': match_expressions_arr}]}}}
+        
+        else:
+            logging.info(src_yaml + " is not a service deploy file! Only support " + str(service_deploy_kind_list))
+            return
+
+    with open(dst_yaml, "w") as fout:
+        yaml.dump(config, fout)
```

Comment:
This section is more serivce template logic specific, suggest to move it to service_template_generate.py. I would suggest file_handler.py should be the place to place pure file operation related functions.

Reply:
Has move this function to service_template_generate.py